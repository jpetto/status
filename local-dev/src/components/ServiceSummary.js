import React, { Component } from 'react';
import { Link } from 'react-router';

class ServiceSummary extends Component {
    render() {
        return (
            <li className="list-group-item">
                <span className={"status " + this.props.status}>{this.props.status}</span>
                {/* soon...seriously <Link to={`/service/${this.props.id}`}>{this.props.name}</Link> */}
                {this.props.name}
            </li>
        );
    }
}

export default ServiceSummary;
