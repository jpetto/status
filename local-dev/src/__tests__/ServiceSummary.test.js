import React from 'react';
import renderer from 'react-test-renderer';

import ServiceSummary from '../components/ServiceSummary';

test('service summary', () => {
    const component = renderer.create(
        <ServiceSummary status='healthy' name='police cops'/>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
