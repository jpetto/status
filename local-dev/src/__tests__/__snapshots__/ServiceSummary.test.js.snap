exports[`test service summary 1`] = `
<li
  className="list-group-item">
  <span
    className="status healthy">
    healthy
  </span>
  police cops
</li>
`;
